PR: https://github.com/odoo/odoo/pull/38694

From: ed26e9eafa31dd8bcaea491b2a6dcd1728a90bd0
From: Florian Gilbert
Date: 2019-12-04 13:47:28

Structural Changes: 4
Total Changes: 25

[IMP] account: Various usability improvements
List of improvements:

Invoice/Bill
- Hide "untaxed amount" section when there is no tax.
- Remove tooltips for tax groups field.
- Remove invoice_partner_icon and set this icon in "Vendor" (in a list view)

Chart of Accounts
- Same views for config bar and regulare one
- By default don't show deprecated accounts and add filter to display them.
- Add some optional fields in list view (taxes, tags, ...)

Dashboard opening balance wizard
- The opening move entry date equals to the opening date - 1 day.

Bank Statements
- Set default order for the list view (by date desc and name desc)

Tax report configuration:
Add 3 new columns for the list view (country_id, code, tag_name)
Add 4 new searchs (name, tag_name, code, country_id)

Task ID: 2048225

closes odoo/odoo#38694

Signed-off-by: Laurent Smet <smetl@users.noreply.github.com>

================================= pseudo patch: =================================

--- a/addons/account/models/account.py
+++ b/addons/account/models/account.py
@@ -397,10 +397,10 @@ class AccountAccount(models.Model):
         self._cr.execute("""
             SELECT aml.id
             FROM account_move_line aml
-            WHERE aml.account_id in (%s)
+            WHERE aml.account_id in %s
             AND EXISTS (SELECT 1 FROM account_account_account_journal_rel WHERE account_account_id = aml.account_id)
             AND NOT EXISTS (SELECT 1 FROM account_account_account_journal_rel WHERE account_account_id = aml.account_id AND account_journal_id = aml.journal_id)
-        """, tuple(self.ids))
+        """, [tuple(self.ids)])
         ids = self._cr.fetchall()
         if ids:
             raise ValidationError(_('Some journal items already exist with this account but in other journals than the allowed ones.'))

--- a/addons/account/models/account_bank_statement.py
+++ b/addons/account/models/account_bank_statement.py
@@ -219,7 +219,7 @@ class AccountBankStatement(models.Model):
 
     _name = "account.bank.statement"
     _description = "Bank Statement"
-    _order = "date desc, id desc"
+    _order = "date desc, name desc, id desc"
     _inherit = ['mail.thread']
 
     name = fields.Char(string='Reference', states={'open': [('readonly', False)]}, copy=False, readonly=True)

--- a/addons/account/models/account_move.py
+++ b/addons/account/models/account_move.py
@@ -158,7 +158,8 @@ class AccountMove(models.Model):
     amount_residual_signed = fields.Monetary(string='Amount Due Signed', store=True,
         compute='_compute_amount', currency_field='company_currency_id')
     amount_by_group = fields.Binary(string="Tax amount by group",
-        compute='_compute_invoice_taxes_by_group')
+        compute='_compute_invoice_taxes_by_group',
+        help='Edit Tax amounts if you encounter rouding issues.')
 
     # ==== Cash basis feature fields ====
     tax_cash_basis_rec_id = fields.Many2one(
@@ -236,7 +237,6 @@ class AccountMove(models.Model):
         help="Auto-complete from a past bill.")
     invoice_source_email = fields.Char(string='Source Email', tracking=True)
     invoice_partner_display_name = fields.Char(compute='_compute_invoice_partner_display_info', store=True)
-    invoice_partner_icon = fields.Char(compute='_compute_invoice_partner_display_info', store=False, compute_sudo=True)
 
     # ==== Cash rounding fields ====
     invoice_cash_rounding_id = fields.Many2one('account.cash.rounding', string='Cash Rounding Method',
@@ -1088,13 +1088,9 @@ class AccountMove(models.Model):
             vendor_display_name = move.partner_id.name
             if not vendor_display_name:
                 if move.invoice_source_email:
-                    vendor_display_name = _('From: ') + move.invoice_source_email
-                    move.invoice_partner_icon = '@'
+                    vendor_display_name = _('@From: ') + move.invoice_source_email
                 else:
-                    vendor_display_name = _('Created by: %s') % (move.sudo().create_uid.name or self.env.user.name)
-                    move.invoice_partner_icon = '#'
-            else:
-                move.invoice_partner_icon = False
+                    vendor_display_name = _('#Created by: %s') % (move.sudo().create_uid.name or self.env.user.name)
             move.invoice_partner_display_name = vendor_display_name
 
     @api.depends('state', 'journal_id', 'invoice_date')

--- a/addons/account/models/company.py
+++ b/addons/account/models/company.py
@@ -79,7 +79,7 @@ class ResCompany(models.Model):
     #Fields of the setup step for opening move
     account_opening_move_id = fields.Many2one(string='Opening Journal Entry', comodel_name='account.move', help="The journal entry containing the initial balance of all this company's accounts.")
     account_opening_journal_id = fields.Many2one(string='Opening Journal', comodel_name='account.journal', related='account_opening_move_id.journal_id', help="Journal where the opening entry of this company's accounting has been posted.", readonly=False)
-    account_opening_date = fields.Date(string='Opening Date', related='account_opening_move_id.date', help="Date at which the opening entry of this company's accounting has been posted.", readonly=False)
+    account_opening_date = fields.Date(string='Opening Entry', default=lambda self: fields.Date.today().replace(month=1, day=1), required=True, help="That is the date of the opening entry.")
 
     # Fields marking the completion of a setup step
     # YTI FIXME : The selection should be factorize as a static list in base, like ONBOARDING_STEP_STATES
@@ -331,10 +331,7 @@ class ResCompany(models.Model):
             if not default_journal:
                 raise UserError(_("Please install a chart of accounts or create a miscellaneous journal before proceeding."))
 
-            today = datetime.today().date()
-            opening_date = today.replace(month=int(self.fiscalyear_last_month), day=self.fiscalyear_last_day) + timedelta(days=1)
-            if opening_date > today:
-                opening_date = opening_date + relativedelta(years=-1)
+            opening_date = self.account_opening_date - timedelta(days=1)
 
             self.account_opening_move_id = self.env['account.move'].create({
                 'ref': _('Opening Journal Entry'),
